RecoEgamma/EgammaPhotonAlgos/src/PhotonEnergyCorrector.cc

```diff
@@ -65,15 +69,9 @@ void PhotonEnergyCorrector::init(const edm::EventSetup& theEventSetup) {
   scEnergyErrorFunction_->init(theEventSetup);
   photonEcalEnergyCorrFunction_->init(theEventSetup);
 
-  if (weightsfromDB_) {
-    if (!regressionCorrector_->IsInitialized())
-      regressionCorrector_->Initialize(theEventSetup, w_db_, weightsfromDB_);
-  }
-  if (!weightsfromDB_ && !(w_file_ == "none")) {
-    if (!regressionCorrector_->IsInitialized())
-      regressionCorrector_->Initialize(theEventSetup, w_file_, weightsfromDB_);
+  if (not regressionCorrector_) {
+    regressionCorrector_ = std::make_unique<EGEnergyCorrector>(regressionCorrectorFactory_->build(theEventSetup));
```

Comment:
@Dr15Jones how can `regressionCorrector_` be non existent? It is always initialized in the constructor, see 
https://github.com/cms-sw/cmssw/pull/35151/files#diff-57977e080ee8dc7a3782ed263ff9e13425dca7b2341cc5b20555b376442fe014R59

Reply:
If `weightsfromDB_` is `true` than `regressionCorrector_` is not set in the constructor and will be set here (since it has to wait until the `EventSetup` is ready).

Reply:
> If `weightsfromDB_` is `true` than `regressionCorrector_` is not set in the constructor and will be set here (since it has to wait until the `EventSetup` is ready).

Ah, ok, thank you! My fault, I misread `regressionCorrectorFactory_` as `regressionCorrector_`